HW06.Rmd

```diff
@@ -14,20 +15,47 @@ Steps:
   3. For 27 genes of interest AND your favorite gene, perform a t-test to see if it is differentially expressed between the WT vs H3K27M samples
   4. Create a graphing function and then create a boxplot that graphs expression between the two groups
   
-  
 #### Code
 
-From the RNA-Seq files, you only need the "Name" from one file and the "TPM" column from all the files. TPM stands for "transcripts per million" and is a common unit for normalized expression data. 
-```{r load in data}
-#hint, using apply (specifically sapply) you can read in the data into a list and then bind the columns together. Or you can use a for loop too. 
+``` {r load packages, echo = FALSE, message = FALSE}
+library(readr)
+library(dplyr)
+library(knitr)
+library(tidyverse)
+```
+
+
+From the RNA-Seq files, you only need the "Name" from one file and the "TPM" column from all the files. TPM stands for "transcripts per million" and is a common unit for normalized expression data.
+
+```{r load in data, message = FALSE}
+
+#Defining factors mainly for better control over the order of boxplots later on
+type <- factor(c("H3K27M_mutated", "WT"))
 
-#you only need the 
+#This chunk creates a list of .sf files in specified subfolder, batch loads them in, binds them together and applies "H3K27M_mutated" or "WT" factor to the values
+H3K27M_list <- Sys.glob("RNA_Seq_processed/H3K27M/*.sf")
+H3K27M_data <- lapply(H3K27M_list, function(x){read_tsv(x)})
+H3K27M.df <- do.call("rbind", H3K27M_data) %>%
+  select(Name, TPM) %>%
+  mutate(type = "H3K27M_mutated") %>%
+  rename(gene_id = Name)
+
+WT_list <- Sys.glob("RNA_Seq_processed/WT/*.sf")
+WT_data <- lapply(WT_list, function(x){read_tsv(x)})
+WT.df <- do.call("rbind", WT_data) %>%

```

Comment:
do.call() is another function I'm not familiar with. I checked rdocumentation.org to see if I could learn more -- "do.call constructs and executes a function call from a name or a function and a list of arguments to be passed to it" -- but it's still unclear to me how this is different from using the function itself if the function is in base r. I'm interested to learn more about what the advantages of do.call are over other ways of passing functions and arguments.

Reply:
Hi! That's a very good question and I was puzzled by this as well. In this case if you just use rbind(), the result will be a matrix, in which each element is a file that contains the observations. However the observations aren't merged - each data frame is stored separately as a separate element. What do.call() seems to do in this case is it also extracts the data frames from these files, so that rbind can bind together actual data frames and generate a matrix of files. I hope my explanation isn't too confusing.